src/protocol/RedemptionHandler.sol

```diff
@@ -81,43 +96,75 @@ contract RedemptionHandler is CoreOwnable{
     /// @notice Calculates the total redemption fee as a percentage of the redemption amount.
     function getRedemptionFeePct(address _pair, uint256 _amount) public view returns(uint256){
         //get fee
-        (uint256 feePct,) = _getRedemptionFee(_pair, _amount);
+        (uint256 feePct,,) = _getRedemptionFee(_pair, _amount);
         return feePct;
     }
 
-    function _getRedemptionFee(address _pair, uint256 _amount) internal view returns(uint256, RedeemptionRateInfo memory){
+    function _getRedemptionFee(address _pair, uint256 _amount) internal view returns(uint256 _redemptionfee, RedeemptionRateInfo memory _rdata, uint256 _totalweight){
         (, , , IResupplyPair.VaultAccount memory _totalBorrow) = IResupplyPair(_pair).previewAddInterest();
         
         //determine the weight of this current redemption by dividing by pair's total borrow
         uint256 weightOfRedeem;
         if (_totalBorrow.amount != 0) weightOfRedeem = _amount * PRECISION / _totalBorrow.amount;
 
         //update current data with decay rate
-        RedeemptionRateInfo memory rdata = ratingData[_pair];
+        // RedeemptionRateInfo memory rdata = ratingData[_pair];
+        _rdata = ratingData[_pair];
         
+        _totalweight = totalWeight;
+
         //only decay if this pair has been used before
-        if(rdata.timestamp != 0){
+        if(_rdata.timestamp != 0){
+            //remove current from total (add back after calcs)
+            _totalweight -= _rdata.usage;
+
             //reduce useage by time difference since last redemption
-            uint192 decay = uint192((block.timestamp - rdata.timestamp) * usageDecayRate);
+            uint192 decay = uint192((block.timestamp - _rdata.timestamp) * usageDecayRate);
+
             //set the pair's usage or weight
-            rdata.usage = rdata.usage < decay ? 0 : rdata.usage - decay;
+            _rdata.usage = _rdata.usage < decay ? 0 : _rdata.usage - decay;
         }
         //update timestamp
-        rdata.timestamp = uint64(block.timestamp);
-        
+        _rdata.timestamp = uint64(block.timestamp);
+
+        //check for over usage and apply a fee
+        ///this uses the current pair weight *before* the current redemption is applied but after the time decay is taken into account
+        uint256 overusageFee;
+        if(_totalweight+_rdata.usage > 0){
```

Comment:
Technically don't need the addition of rdata.usage as total weight will be positive as soon as a second pair is redeemed from